nameko/containers.py

```diff
@@ -353,41 +348,55 @@ def spawn_worker(self, entrypoint, args, kwargs,
             self, service, entrypoint, args, kwargs, data=context_data)
 
         _log.debug('spawning %s', worker_ctx)
-        gt = self._worker_pool.spawn(self._run_worker, worker_ctx,
-                                     handle_result)
-        self._active_threads[gt] = entrypoint
-        gt.link(self._handle_thread_exited)
+        gt = self._worker_pool.spawn(
+            self._run_worker, worker_ctx, handle_result
+        )
+        self._worker_threads[worker_ctx] = gt
+        self.worker_created(worker_ctx)
```

Comment:
probably want to call `link` before the `worker_created` hook in case the latter yields